sechub-scan/src/main/java/com/daimler/sechub/adapter/DefaultExecutorConfigSupport.java

```diff
@@ -0,0 +1,111 @@
+import com.daimler.sechub.domain.scan.product.config.ProductExecutorConfigSetupJobParameter;
+import com.daimler.sechub.sharedkernel.SystemEnvironment;
+import com.daimler.sechub.sharedkernel.validation.AssertValidation;
+import com.daimler.sechub.sharedkernel.validation.Validation;
+
+/**
+ * A standard executor configuration support. Supports environment entry evaluation, simple key value checks etc.
+ * 
+ * @author Albert Tregnaghi
+ *
+ */
+public class DefaultExecutorConfigSupport {
+
+    private static final String ENV_PREFIX_ID = "env:";
+
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultExecutorConfigSupport.class);
+
+    protected Map<String, String> configuredExecutorParameters = new TreeMap<>();
+
+    protected ProductExecutorConfig config;
+    private SystemEnvironment systemEnvironment;
+
+    public DefaultExecutorConfigSupport(ProductExecutorConfig config, SystemEnvironment systemEnvironment, Validation<ProductExecutorConfig> validation) {
+        notNull(config, "config may not be null!");
+        notNull(systemEnvironment, "systemEnvironment may not be null!");
+
+        this.config = config;
+        this.systemEnvironment = systemEnvironment;
+
+        if (validation != null) {
+            AssertValidation.assertValid(config, validation);
+        }
+
+        /* create a simple map containing parameters */
+        List<ProductExecutorConfigSetupJobParameter> jobParameters = config.getSetup().getJobParameters();
+        for (ProductExecutorConfigSetupJobParameter jobParameter : jobParameters) {
+            configuredExecutorParameters.put(jobParameter.getKey(), jobParameter.getValue());
+        }
+
+    }
+    
+    public String getPasswordOrAPIToken() {
+        ProductExecutorConfigSetupCredentials credentials = config.getSetup().getCredentials();
+        String pwd = credentials.getPassword();
+        return evaluateEnvironmentEntry(pwd);
+    }
+
+    public String getProductBaseURL() {
+        return config.getSetup().getBaseURL();
+    }
+
+    public String getUser() {
+        ProductExecutorConfigSetupCredentials credentials = config.getSetup().getCredentials();
+        String user = credentials.getUser();
+        return evaluateEnvironmentEntry(user);
+    }
+
+    private String evaluateEnvironmentEntry(String data) {
+        if (data == null) {
+            return null;
+        }
+        if (!data.startsWith(ENV_PREFIX_ID)) {
+            return data;
+        }
+        String key = data.substring(ENV_PREFIX_ID.length());
+        String value = systemEnvironment.getEnv(key);
+        if (value == null) {
+            LOG.warn("No environment entry defined for variable:{}", key);
+        }
+        return value;
+    }
+
+    protected boolean getParameterBooleanValue(String key) {
+        String asText = getParameter(key);
+        return Boolean.parseBoolean(asText);
+    }
+
+    protected String getParameter(String key) {
+        if (key == null) {
```

Comment:
By changing the logic, you can avoid one return and make it more readable: 

~~~
String parameter = null;

if (key != null) {
  parameter = configuredExecutorParameters.get(key);
}

return parameter;

Reply:
I prefer my style.